contributors/design-proposals/federated-statefulsets.md

```diff
@@ -0,0 +1,249 @@
+This cannot work across clusters, as local pod IPs cannot be reached across the clusters.
+
+The proposal to make it work across the clusters is to assign a service type &#39;LoadBalancer&#39; for each pod instance that is created in the clusters.
+The service name will be based on the ordinal pod name assigned to each pod.
+Additionally a set of label selectors will be added to each individual pod to ensure that the cluster routes the specific service traffic to the specific pod.
+The service type &#39;LoadBalancer&#39; will ensure that the data traffic can be routed across clusters.
+The network dns names for each pod will remain very similar to the network names used by the in-cluster statefulsets.
+
+As an example:
+
+For a federation with 3 clusters all in the same DNS zone _federation.example_ and the statefulset name _web_ the dns names for instances might look like:
+```
+web-0.mynamespace.myfederation.svc.federation.example
+web-1.mynamespace.myfederation.svc.federation.example
+Web-2.mynamespace.myfederation.svc.federation.example
+```
+
+The difference compared to an in-cluster dns name for an instance of a statefulset can be noted where the dns id starts with **$(podname).$(governing service domain)**, whereas when federated, in this approach, it will start only with **$(service name).**
+This will be achieved by federated controller creating and monitoring individual services in specific clusters (not a federated service) against each pod, with the same name as that pod&#39;s ordinal identity.
+It will further need to be ensured that the pods be assigned the same hostname as the pod name (ordinal identity) and service name to keep it consistent with the in-cluster statefulsets topology.
+
+There is one interesting catch about the service creation, here.
+The service needs to exist locally in the cluster (1 service per replica pod across clusters), which is unlike a federated service.
+If we use the existing federated service mechanism, to create this service, and the service with a particular name (for example web-0 in example above) is created in federation, it will propagate the same to all federated clusters.
+This is unnecessary for the solution mentioned in this section (or for further reading in the following section), and would be extreme waste of resources.
+Some simple proposals to tackle this problem are mentioned at the end of this document under section &quot; **Handling additional LB Services**&quot;.
+
+## Design Alternative 2
+
+In this approach, the federated statefulset controller will behave quite similar to federated replicaset or the federated deployment controller.
+The federated controller would create and monitor individual statefulsets (rather then pods directly) partitioning and distributing the total stateful replicas across the federated clusters.
+
+As a proposal in this design we suggest the possibility of the pods having multiple identities.
```

Comment:
No it doe not. In a normal in-cluster statefulset a stateful pod gets 2 identities:
- the dns name accessible within the cluster
- the hostname visible to the stateful app

The suggestion in this design retains both and then adds another dns name accessible across the cluster. I specify the additional dns name as another identity, thus multiple identities. I dont know of any such use case as of now, but if needed the same design can be extended to have even more dns names visible locally or globally reflecting more identities.